File: output/llama3.1-8b/avatar/Python/Java/atcoder_ABC125_C.java
public class atcoder_ABC125_C {
    public static int gcd(int a, int b) {
        if (b == 0)
            return a;
        return gcd(b, a % b);
    }

    public static void main(String[] args) {
        java.util.Scanner scanner = new java.util.Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        String[] input = scanner.nextLine().split(" ");
        int[] array = new int[n];
        for (int i = 0; i < n; i++) {
            array[i] = Integer.parseInt(input[i]);
        }
        int[] left = new int[n + 1];
        int[] right = new int[n + 1];
        left[0] = 0;
        for (int i = 0; i < n; i++) {
            left[i + 1] = gcd(left[i], array[i]);
        }
        for (int i = n - 1; i >= 0; i--) {
            right[i] = gcd(right[i + 1], array[i]);
        }
        int ans = 0;
        for (int i = 0; i < n; i++) {
            ans = Math.max(gcd(left[i], right[i + 1]), ans);
        }
        System.out.println(ans);
    }
}

File: output/llama3.1-8b_IO_1/avatar/Java/Python/atcoder_ABC125_C.py
import math

A = []
N = int(input())
for i in range(N):
    line = input()
    A.append(int(line.split()[0]))

M = [0] * N
L = [0] * N
R = [0] * N
L[0] = A[0]
for i in range(1, N):
    L[i] = math.gcd(L[i-1], A[i])

R[N-1] = A[N-1]
for i in range(N-2, -1, -1):
    R[i] = math.gcd(R[i+1], A[i])

answer = max(L[-2], R[1])
for i in range(1, N-1):
    answer = max(answer, math.gcd(L[i-1], R[i+1]))

print(answer)
